velox/functions/lib/Re2Functions.cpp

```diff
@@ -412,6 +425,47 @@ class OptimizedLikeWithMemcmp final : public VectorFunction {
       vector_size_t reducedPatternLength)
       : pattern_{pattern}, reducedPatternLength_{reducedPatternLength} {}
 
+  OptimizedLikeWithMemcmp(
+      StringView pattern,
+      vector_size_t reducedPatternLength,
+      std::vector<std::string> fixedPatterns)
+      : pattern_{pattern},
+        reducedPatternLength_{reducedPatternLength},
+        fixedPatterns_{fixedPatterns} {}
+
+  // Match all fixed patterns in an input string having fixed patterns
+  // interspersed with one or more '%' wildcard character streams.
+  bool simplifiedPatternMatch(const StringView input) const {
+    auto startPosition = 0;
+    auto inputString = input.str();
+    auto patternString = pattern_.str();
+    auto noTrailingWildcard = (patternString.back() != '%');
```

Comment:
Why do you need this special case for noTrailingWildcard ?

Reply:
This case is added to ensure input-pattern pairs such as "like(abcabcdef, abc%abc)" return false.